### ISSUE TITLE
Absolute (hard-coded?) paths in web admin

### ISSUE THREAD

**Issue by 9662** [processone/ejabberd#3043]
My report is a duplicate of #1449, but since the closing comment appears to solicit feedback on usage / desirability of this feature, I thought it proper to open this ticket (I am not allowed to write a comment on #1449).

**Comment by 9662** [comment]
It's a bit worse than I thought actually. It is not just that the links are absolute. As far as I can read the source (I'm not an Erlang programmer), they are [actually *hard-coded*](https://github.com/processone/ejabberd/blob/c49edaca1964cf525db1c4dac9f6aacc1bc39014/src/ejabberd_web_admin.erl#L342)?

**Comment by badlop** [comment]
With that recent change, you can configure things like those:

```
listen:
  -
    port: 5443
    module: ejabberd_http
    tls: true
    request_handlers:
      "/xyz/aaa/momomo": ejabberd_web_admin
  -
    port: 5280
    module: ejabberd_http
    request_handlers:
      "/abc/def/ghi/toto": ejabberd_web_admin
```

**Comment by mremond** [comment]
@badlop What should we do with this issue ? Is it solved ?

**Comment by badlop** [comment]
Ah yes, this got implemented in October and released in 19.11.

**Comment by dkuku** [comment]
I think I got similar problem in 20.12
I installed ejabberd in an elixir umbrella app using @mremond post from ![medium](https://medium.com/processone/how-to-use-ejabberd-as-an-elixir-application-dependency-e31dcea55e7a)
and when loading the admin page I got no styling and all links are returning 404. I searched a bit but no luck and then I tried to add admin in between my host and the path and everything started to work with proper styling

Links on the admin page are pointing to `WEBSERVER/muc` and these should point to `WEBSERVER/admin/muc` 
![ejabberd](https://user-images.githubusercontent.com/904179/103904820-b5120a00-50f5-11eb-9485-f4eab83e822d.png)
example screenshot from `/admin` route

**Comment by dkuku** [comment]
I managed to fix it by including in html head 
`<base href="/admin/" target="_self">`
![image](https://user-images.githubusercontent.com/904179/103906564-028f7680-50f8-11eb-91fd-581e21fc7195.png)

**Comment by badlop** [comment]
Probably the problem is that you visit http://localhost:5280/admin without the trailing slash. This got fixed in 5ec214386e3141e000fbcfaa53928c48f132c8fe by redirecting such URLs to the correct one:

```
GET /admin HTTP/1.1
User-Agent: Wget/1.20.1 (linux-gnu)
Accept: */*
Accept-Encoding: identity
Host: localhost:5280
Connection: Keep-Alive

HTTP/1.1 301 Moved Permanently
Content-Type: text/html; charset=utf-8
Content-Length: 0
Location: /admin/
```

And I wonder how can you achieve to still produce the problem. It works correctly for me, using ejabberd 20.12 compiled from source code.
